source/message.html

```diff
@@ -4,24 +4,82 @@
 <TITLE></TITLE>
 <SCRIPT LANGUAGE=javascript>
 <!--
-function escapeKeyPress(e) {
+// Though these should be constants, MSHTML does not support const as of 2024
+var args = window.dialogArguments;
+var copyStatusTimeout = 5000;  // milliseconds before copy status message goes away
+
+function handleKeyPress(e) {
 	e = e || window.event;
 	if(e.keyCode == 27){ // code for escape
 		window.close();
+	} else if (e.altKey && e.keyCode == 67) { // code for Alt+C
+		onCopyButtonPress();
+	}
+}
+
+function onCopyButtonPress() {
+	// Copy code came from http://www.codestore.net/store.nsf/unid/DOMM-4QHQE8/
+	var rng = document.body.createTextRange();
+	rng.moveToElementText(messageID);
+	rng.scrollIntoView();
+	rng.select();
+	var success = rng.execCommand("Copy");
+	rng.collapse(false);
+	rng.select();
+	if (args) {
+		if (success) {  // Notify the user about the copy result
+			copyStatusDiv.innerHTML = '<p id="copyStatusText"><strong>' + args.item('copySuccessfulAlertText') + '</strong></p>';
+		} else {
+			copyStatusDiv.innerHTML = '<p id="copyStatusText"><strong>' + args.item('copyFailedAlertText') + '</strong></p>';
+		}
+		// Time out the user alert message
+		setTimeout(function() {
+			copyStatusDiv.innerHTML = '';
+		}, copyStatusTimeout);
 	}
-};
+}
+
+function onCloseButtonPress() {
+	window.close();
+}
 
 function windowOnLoad() {
-	var args = window.dialogArguments;
 	if (args) {
 		document.title = args.item('title');
 		messageID.innerHTML = args.item('message');
+		// If caller wants a close button
+		if (args.item('closeButtonText') != null) {
+			closeButton.innerHTML = args.item('closeButtonText');  // Assign the (translated) label
+			closeButton.style.display = 'inline';  // Display the button
+			buttonDiv.style.display = 'block';  // Display the button section
+		}
+		// If caller wants a copy to clip button
+		if (args.item('copyButtonText') != null) {
+			copyButton.innerHTML = args.item('copyButtonText');  // Assign the (translated) label
+			copyButtonSpan.style.display = 'inline';  // Display the button
+			copyStatusDiv.style.display = 'block';  // Display the copy status
+			buttonDiv.style.display = 'block';  // Display the section
+			if (args.item('copyButtonAcceleratorAccessibilityLabel') != null) {  // If translated text for the accelerator is provided
+				copyButtonAcceleratorAccessibilityLabel.innerHTML = args.item('copyButtonAcceleratorAccessibilityLabel');
+			}
+		}
 	}
 } 
 //-->
 </SCRIPT>
 </HEAD>
-<BODY tabindex='0' id='main_body' style="margin:1em" LANGUAGE=javascript onload="return windowOnLoad()" onkeypress="return escapeKeyPress()" >
+<body tabindex='0' id='main_body' style="margin:1em" LANGUAGE=javascript onload="return windowOnLoad()" onkeydown="return handleKeyPress()" >
```

Comment:
I don't think this language attribute is valid, if anything it should use the NVDA language code

```suggestion
<body tabindex='0' id='main_body' style="margin:1em" onload="return windowOnLoad()" onkeydown="return handleKeyPress()" >
```

Reply:
That's been there for a very long time. No idea when it was valid, if ever.